### PR TITLE
Allow to pass options for deploy keys

### LINKED ISSUE #396: Allow deploy keys to set the can_push attribute.
## The problem

https://docs.gitlab.com/ee/api/deploy_keys.html#add-deploy-key

We do not allow users to set the `can_push` attribute via the current API.

## The fix

Allow any extra parameters to be passed when deploying a key (optional `options={}` hash to the method).

### PR DESCRIPTION
**Pull request by born4new**
Fixes #396

### CODE REVIEW

File: spec/gitlab/client/projects_spec.rb [hunk 1]
```diff
@@ -515,6 +515,40 @@
...
+    end
+
+    context 'some options' do
+      before do
+        stub_post("/projects/42/deploy_keys", "project_key")
+        @deploy_key = Gitlab.create_deploy_key(42, 'My Key', 'Key contents', can_push: true)
+      end
+
+      it "gets the correct resource" do
+        expect(a_post("/projects/42/deploy_keys").
```
Comment (NARKOZ) [inline spec/gitlab/client/projects_spec.rb]: You missed a bracket there

Comment (born4new) [reply spec/gitlab/client/projects_spec.rb]: Oh indeed: will fix.

**Comment by born4new** [comment]
@NARKOZ Done.

File: spec/gitlab/client/projects_spec.rb [hunk 2]
```diff
@@ -515,6 +515,40 @@
...
+
+    context 'some options' do
+      before do
+        stub_post("/projects/42/deploy_keys", "project_key")
+        @deploy_key = Gitlab.create_deploy_key(42, 'My Key', 'Key contents', can_push: true)
+      end
+
+      it "gets the correct resource" do
+        expect(a_post("/projects/42/deploy_keys")).
+          with(body: { title: 'My Key', key: 'Key contents', can_push: true }).to have_been_made
```
Comment (NARKOZ) [inline spec/gitlab/client/projects_spec.rb]: Mismatched parentheses:

```
expect(a_post("/projects/42/deploy_keys").
  with(body: { title: 'My Key', key: 'Key contents', can_push: true })).to have_been_made
```

Comment (born4new) [reply spec/gitlab/client/projects_spec.rb]: Very sorry...This time, I launched the specs locally.

**Comment by NARKOZ** [comment]
@born4new no problem. Thank you! 👍

**Comment by born4new** [comment]
@NARKOZ My pleasure. Would it be possible for you to create a gem release out of it? It's just so that people can just rely on the rubygems version of gitlab-ruby.

**Comment by NARKOZ** [comment]
@born4new version 4.5.0

**Pull request merged**